### ISSUE TITLE
pen arm not smooth

### ISSUE THREAD

**Issue by geauxgetter** [cocktailyogi/EggDuino#24]
So is it normal for the pen arm to jerk around? I can't seem to get a nice fluid movement. I'm using 16th micro steps on my cnc shield and tried running my timings are 300, 200 and 100 but nothing really helps. I've also adjusted my motor current and that didn't improve anything either. I'm wondering how people are able to get nice smooth lines without the "stepping". Running version 1.6a with inkscape .92

**Comment by pyrophreek** [comment]
What stepper chip are you using? How many steps per rotation does your motor have? calculate the distance each step should travel, and if that is what you are seeing as stepping, then you are at the limit of your setup.  If your stepper board supports replacing the drivers, (step sticks) try replacing with 32x microstepping drv8825 drivers, or even 256 microstepping TMC2100 based boards.

**Comment by geauxgetter** [comment]
stepper is the a4988. I am getting 3200 steps per rotation. I'm using the UNO with a CNC shield.

**Comment by PainHealer** [comment]
My setups is exactly the same and I have the same issue. Both steppers are running very "rough" and produce steps/stairs instead of straight lines or "round curves". I also have 3200 (micro)steps for one full rotation, so 1/16 microstepping basically seems to work wirth my 1.8 degree steppers. I don't think that going to 1/32 or even 1/256 microstepping will lead anywhere or make things better. Can anybody confirm that this setup with this arduino sketch can even do higher quality prints?

**Comment by geauxgetter** [comment]
I changed the drivers for 1/32 steps and it's perfectly smooth now. Only
problem is I have to resize all inkscape plots to 6400x1600

**Comment by PainHealer** [comment]
Thanks for xour reply, did you change parameters for the microstepping in the sketch to 32 or leave it at 16? I just swapped drivers from A4988 to DRV8825 and changed it to 32 and now the steppers are not turning at all. :-\

**Comment by geauxgetter** [comment]
The drv8825 goes with the pot facing to the top. Not the same way as the
4988. Only thing I did was change the inkscape document size to 6400x1600
and boom, perfect lines

**Comment by PainHealer** [comment]
I had to change the settings in the firmware back to 16, the drivers already had the correct orrientation, but thanks for the advice anyways, that often gets mixed up. While the overall line quality has improved and rotation is very smooth now, the pen-arm still "snaps" like if there is a "grid" underneath the coordinates. It looks and sounds as if it physically still does only full 1.8 degree steps, even if it needs 6400 microsteps per full rotation. I'm just not getting what could be wrong.

**Comment by geauxgetter** [comment]
I had to tweak the voltage on the drivers to get smooth movements. It was
very picky.

**Comment by PainHealer** [comment]
You are right, thank you very much! When I swapped the drivers I adjusted the reference voltage very loosely somewhere between "it (the stepper) won't move" and "it just does not get hot", in volts this was something around 0.250-0.30, now I'm at 0.185 volts and it does not get any better. The pen moves smoothly over 80% of the surface from left to rigth and does only get a little bit "shaky" when it slows down when printing circles or elipses.

**Comment by geauxgetter** [comment]
Make sure the egg motor is at the same voltage.. I noticed that also caused
my prints to be less than perfect if not dialed in right. Glad I could help!

**Comment by Jantleing** [comment]
Main issue is the communication speed...  instructions are not sent fast enough for smooth movements at higher speed.
Communication at 115200 baud eg gives smooth movement even at 1000 steps/s

Open port  in Eggduino.ino  like this:
Serial.begin(115200);

and for inkscape extension change following line within ebb_serial.py  from
  serial_port = serial.Serial(port_name, timeout=2.0)  
to
  serial_port = serial.Serial(port_name, timeout=2.0, baudrate=115200)
